src/main/java/wordsWithSpaces/WordsWithSpaces.java

```diff
@@ -0,0 +1,20 @@
+package wordsWithSpaces;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class WordsWithSpaces {
+    private static Pattern pattern = Pattern.compile("[A-Za-z]\\S*\\b");
```

Comment:
Basically, you're finding words and collecting them to a list. After that list values are joined with a space.
You can use much simplier regexp: `\\w+` which will find just a single word.

One question: how should it work for `words.getWords("   some,         thing   ")` ? Should it return `some thing` or `some, thing`?

Reply:
It should return `some thing` because according to the task we need to have a string as a result

Reply:
Oh yes, '\w+\ is much simpler... updated